### PR TITLE
Featurelistsummary and CCS

### PR DESCRIPTION
**Pull request by Tarush-Singh35**
@robinschmid Fix for the Issue #608 and CCS

### CODE REVIEW

**Comment by Tarush-Singh35** [comment]
@robinschmid @SteffenHeu commits over here please review the code sorry for the troubles 😄

File: src/main/java/io/github/mzmine/modules/dataprocessing/filter_isotopefinder/IsotopeFinderTask.java [hunk 1]
```diff
@@ -192,11 +195,25 @@ public void run() {
...
       if (scanRange == ScanRange.SINGLE_MOST_INTENSE) {
         // add isotope pattern and charge
         feature.setIsotopePattern(pattern);
         feature.setCharge(bestCharge);
+        //Final CCS Calculation
+        RawDataFile data = feature.getRawDataFile();
+        Float mobility = feature.getMobility();
+        MobilityType mobilityType = feature.getMobilityUnit();
+        if (data instanceof IMSRawDataFile imsfile) {
+          if (imsfile.getCCSCalibration() != null) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/filter_isotopefinder/IsotopeFinderTask.java:207]: This prevents the CCS calculation for the TIMS mobility type, which is already pre-calibrated. I think the best solution is to refactor the CCSUtils.hasValidMobilityType(IMSRawDataFile) to check for the calibration and the mobility type so we don't do double checks. Then remove this line 207.

File: src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java [hunk 2]
```diff
@@ -75,6 +76,16 @@ public static Float calcCCS(double mz, @NotNull Float mobility,
     };
   }
 
+  public static boolean hasValidMobilityType(@NotNull IMSRawDataFile file) {
+    //Valid Mobility Check for CCS calculation in the function
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java:80]:
```
if(file.getMobilityType() == TIMS) { return true;}
else if ((file.getMobilityType() == MobilityType.DRIFT_TUBE
        || file.getMobilityType() == MobilityType.TRAVELING_WAVE) && file.getCCSCalibration() != null) {
      return true;
}
return false;
```

File: src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java [hunk 3]
```diff
@@ -75,6 +76,16 @@ public static Float calcCCS(double mz, @NotNull Float mobility,
     };
   }
 
+  public static boolean hasValidMobilityType(@NotNull IMSRawDataFile file) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java:79]: add javadoc here to explain what the method does

File: src/main/java/io/github/mzmine/gui/mainwindow/FeatureListSummaryController.java [hunk 4]
```diff
@@ -194,4 +209,37 @@ private void setSelectionToLastBatchStep(ParameterSet parameters) {
...
+    try {
+      BufferedWriter writer = Files.newBufferedWriter(file.toPath(), StandardCharsets.UTF_8);
+      PrintWriter pw = new PrintWriter(writer);
+      for (FeatureListAppliedMethod item : lvAppliedMethods.getItems()) {
+        StringBuilder sb = new StringBuilder(item.getDescription());
+        pw.println(sb);
+        ParameterSet parameterSet = item.getParameters();
+        for (Parameter<?> parameter : parameterSet.getParameters()) {
+          pw.println(parameterToString(parameter));
+        }
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/gui/mainwindow/FeatureListSummaryController.java:237]: Add a new line after printing all parameters to increase the readability.

**Review by SteffenHeu — COMMENTED** [review]
Hey Tarush, I looked over it and it looks great so far. I found one small bug in the CCS calculation that needs to be fixed and a comment for the export to increase the readability.
Afterwards we can merge it

File: src/main/java/io/github/mzmine/modules/dataprocessing/filter_isotopefinder/IsotopeFinderTask.java (hunk 1 above)
Comment (Tarush-Singh35) [reply src/main/java/io/github/mzmine/modules/dataprocessing/filter_isotopefinder/IsotopeFinderTask.java:207]: Yes I will fix this Thank you for your insight

File: src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java (hunk 2 above)
Comment (Tarush-Singh35) [reply src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java:80]: Thanks For the Help

File: src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java (hunk 3 above)
Comment (Tarush-Singh35) [reply src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java:79]: I will surely do that

File: src/main/java/io/github/mzmine/gui/mainwindow/FeatureListSummaryController.java (hunk 4 above)
Comment (Tarush-Singh35) [reply src/main/java/io/github/mzmine/gui/mainwindow/FeatureListSummaryController.java:237]: i will fix this thanks for the Insight

File: src/main/java/io/github/mzmine/modules/dataprocessing/filter_isotopefinder/IsotopeFinderTask.java (hunk 1 above)
Comment (Tarush-Singh35) [reply src/main/java/io/github/mzmine/modules/dataprocessing/filter_isotopefinder/IsotopeFinderTask.java:207]: done

File: src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java (hunk 2 above)
Comment (Tarush-Singh35) [reply src/main/java/io/github/mzmine/modules/dataprocessing/id_ccscalc/CCSUtils.java:80]: done

File: src/main/java/io/github/mzmine/gui/mainwindow/FeatureListSummaryController.java (hunk 4 above)
Comment (Tarush-Singh35) [reply src/main/java/io/github/mzmine/gui/mainwindow/FeatureListSummaryController.java:237]: [Export.csv](https://github.com/mzmine/mzmine3/files/8408188/Export.csv)
is this okay @SteffenHeu ?

Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/gui/mainwindow/FeatureListSummaryController.java:237]: Yes, much better!

Comment (Tarush-Singh35) [reply src/main/java/io/github/mzmine/gui/mainwindow/FeatureListSummaryController.java:237]: Okay thanks

**Comment by Tarush-Singh35** [comment]
@SteffenHeu i made the commits please Check

**Comment by SteffenHeu** [comment]
Looks good, thanks!

**Comment by Tarush-Singh35** [comment]
> Looks good, thanks!

Thanks for helping 😄 and sorry for the troubles.

**Pull request merged**